### PR TITLE
MRG Better docs

### PR DESCRIPTION
**Pull request by jasmainak**
partially addresses: https://github.com/mne-tools/mne-python/issues/1495

don't merge. I have more stuff to add. Suggestions welcome

### CODE REVIEW

**Comment by agramfort** [comment]
thanks heaps @jasmainak for taking a stab at this

this is very much appreciated

**Comment by jasmainak** [comment]
:) do you know if there is a way to cross-reference specific examples or tutorials from the manual?

**Comment by jasmainak** [comment]
okay, I think I'm done. Ready for review / merge. Need to get back to my PhD ;) Will make more improvements when I'm bored during Christmas :)

File: doc/advanced_setup.rst [hunk 1]
```diff
@@ -106,6 +106,19 @@ report them on the MNE mailing list or on github to get assistance.
 
 .. _inside_martinos:
```
Comment (agramfort) [inline doc/advanced_setup.rst]: this should be just above the Inside martinos section

File: doc/advanced_setup.rst [hunk 2]
```diff
@@ -106,6 +106,19 @@ report them on the MNE mailing list or on github to get assistance.
...
 
+Installing Mayavi
+^^^^^^^^^^^^^^^^^
+
+In our experience, the easiest way to install `mayavi` is using Anaconda. Just do::
+
+    conda install mayavi
```
Comment (agramfort) [inline doc/advanced_setup.rst]: if you use Anaconda. You can point to:

http://docs.enthought.com/mayavi/mayavi/installation.html

File: doc/manual/cookbook.rst [hunk 3]
```diff
@@ -1,9 +1,5 @@
 .. _cookbook:
 
-========
-Cookbook
-========
```
Comment (agramfort) [inline doc/manual/cookbook.rst]: why ? this will break links to cookbook_

File: doc/manual/datasets_index.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
+
```
Comment (agramfort) [inline doc/manual/datasets_index.rst]: you should add a tag

.. _datasets:

and a title. So you can crossref

File: doc/manual/datasets_index.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+into the left and right visual field, interspersed by tones to the
+left or right ear. The interval between the stimuli was 750 ms. Occasionally
+a smiley face was presented at the center of the visual field.
+The subject was asked to press a key with the right index finger
+as soon as possible after the appearance of the face. To fetch this dataset, do::
+
+    from mne.datasets import sample
+    data_path = sample.data_path()  # returns the folder in which the data is locally store
+
+Once the ``data_path`` is known, it's contents can be examined and the data can be loaded using :ref:`IO functions <ch_convert>`.
```
Comment (agramfort) [inline doc/manual/datasets_index.rst]: can you avoid the long lines?

thx

File: doc/manual/datasets_index.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+Brainstorm
+##########
+Dataset fetchers for three Brainstorm tutorials are available. Users must agree to the
+license terms of these datasets before downloading them. These files are recorded in a CTF 275 system.
+The data is converted to `fif` format before being made available to MNE users. However, MNE-Python now supports
+IO for the `ctf` format as well in addition to the C converter utilities. Please consult the :ref:`IO section <ch_convert>` for details.
```
Comment (agramfort) [inline doc/manual/datasets_index.rst]: I think we should share now the native ds files.

Comment (agramfort) [inline doc/manual/datasets_index.rst]: ping @Eric89GXL

File: doc/manual/datasets_index.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+Auditory
+========
+To access the data, use the following python commands::
+    
+    from mne.datasets.brainstorm import bst_raw
+    data_path = bst_raw.data_path()
```
Comment (agramfort) [inline doc/manual/datasets_index.rst]: is this rendered as code?

even without the >>> ?

**Comment by agramfort** [comment]
does renaming datasets.rst break any link?

you should make this page accessible from the dataset_index page

also you should add link to the main examples that use these datasets.

something like

:ref:`example_datasets_plot_xxx.py` 

should do the job

thx heaps !

File: doc/manual/datasets_index.rst (hunk 7 above)
Comment (jasmainak) [inline doc/manual/datasets_index.rst]: yeah, it's rendered as code. It's okay without the >>>

File: doc/advanced_setup.rst [hunk 8]
```diff
@@ -103,6 +103,29 @@ runtime accordingly.
 If you use another Python setup and you encounter some difficulties please
 report them on the MNE mailing list or on github to get assistance.
 
+Installing Mayavi
+^^^^^^^^^^^^^^^^^
+
+If you have Anaconda installed (recommended), the easiest way to install `mayavi` is to do::
+
+    conda install mayavi
```
Comment (Eric89GXL) [inline doc/advanced_setup.rst]: one reason to add `>>>` to the Python code snippets is the same reason you probably should add a `$` prefix to these lines -- to make it clearer which language is being used.

Comment (Eric89GXL) [inline doc/advanced_setup.rst]: (even if we omit `>>>` for Python code, we should at least put the `$` here so people have more indication that it's not Python)

File: doc/advanced_setup.rst [hunk 9]
```diff
@@ -103,6 +103,29 @@ runtime accordingly.
 If you use another Python setup and you encounter some difficulties please
 report them on the MNE mailing list or on github to get assistance.
 
+Installing Mayavi
+^^^^^^^^^^^^^^^^^
+
+If you have Anaconda installed (recommended), the easiest way to install `mayavi` is to do::
```
Comment (Eric89GXL) [inline doc/advanced_setup.rst]: The first line sentence should probably mention that Mayavi is only available for Python2.7, so if you're running 3.3+, you can't install it

File: doc/manual/datasets_index.rst [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+In this experiment, checkerboard patterns were presented to the subject
+into the left and right visual field, interspersed by tones to the
+left or right ear. The interval between the stimuli was 750 ms. Occasionally
+a smiley face was presented at the center of the visual field.
+The subject was asked to press a key with the right index finger
+as soon as possible after the appearance of the face. To fetch this dataset, do::
+
+    from mne.datasets import sample
+    data_path = sample.data_path()  # returns the folder in which the data is locally store
```
Comment (Eric89GXL) [inline doc/manual/datasets_index.rst]: stored

File: doc/manual/datasets_index.rst [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+IO for the `ctf` format as well in addition to the C converter utilities. Please consult the :ref:`IO section <ch_convert>` for details.
+
+Auditory
+^^^^^^^^
+To access the data, use the following python commands::
+    
+    from mne.datasets.brainstorm import bst_raw
+    data_path = bst_raw.data_path()
+
+Further details can be found at the `auditory dataset tutorial`_ on the Brainstorm website. An example replicating this tutorial can be found at :ref:`example_datasets_plot_brainstorm_data.py`
```
Comment (Eric89GXL) [inline doc/manual/datasets_index.rst]: partially replicating

File: doc/manual/datasets_index.rst [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+Resting state
+^^^^^^^^^^^^^
+To access the data, use the python command::
```
Comment (Eric89GXL) [inline doc/manual/datasets_index.rst]: I think Python is supposed to be captalized

File: doc/manual/datasets_index.rst [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+The recordings were made using the BCI2000 system. To load a subject, do::
+
+    from mne.io import concatenate_raws, read_raw_edf
+    from mne.datasets import eegbci
+    raw_fnames = eegbci.load_data(subject, runs)
+    raws = [read_raw_edf(f, preload=True) for f in raw_fnames]
+    raw = concatenate_raws(raws)
+
+
+Do not hesitate contacting MNE-Python developers on the `MNE mailing list`_ to discuss the possibility to add more publicly available datasets.
```
Comment (Eric89GXL) [inline doc/manual/datasets_index.rst]: Do not hesitate to contact

File: doc/manual/memory.rst [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+^^^
+MNE-Python can read data on-demand using the ``preload`` option provided in :ref:`IO functions <ch_convert>`. For example::
+
+    from mne import io
+    from mne.datasets import sample
+    data_path = sample.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+    raw = io.Raw(raw_fname, preload=False)
+
+By default, most IO functions load data in chunks of approximately ``100 MB``.
```
Comment (Eric89GXL) [inline doc/manual/memory.rst]: I don't think users need to know this.

File: doc/manual/memory.rst [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    events = mne.find_events(raw)
+    event_id, tmin, tmax = 1, -0.2, 0.5
+    picks = mne.pick_types(raw.info, meg=True, eeg=True, stim=False, eog=True)
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks, baseline=(None, 0), reject=dict(eeg=80e-6, eog=150e-6),
+                        preload=False)
+
+When ``preload=False``, the epochs data is loaded from the disk on-demand. Note that ``preload=False`` for epochs will work even if the ``raw`` object
+has been loaded with ``preload=True``. Preloading is also supported for :func:`mne.read_epochs`.
+
+.. warning:: This comes with a caveat. When ``preload=False``, data rejection based on peak-to-peak thresholds is executed when the data is loaded from disk, `not` when the `raw` object is created.
```
Comment (Eric89GXL) [inline doc/manual/memory.rst]: emphasis for "not" should be done using asterisks not backticks

File: doc/tutorials/report.rst [hunk 16]
```diff
@@ -56,6 +56,12 @@ To generate the report in parallel::
     mne report --path MNE-sample-data/ --info MNE-sample-data/MEG/sample/sample_audvis-ave.fif \ 
         --subject sample --subjects-dir MNE-sample-data/subjects --verbose --jobs 6
 
+The report rendered on sample-data is shown below:
+
+
+    .. image:: mne-report.png
```
Comment (Eric89GXL) [inline doc/tutorials/report.rst]: Ouch, 430 kB file added to the repo. Any way around this? Maybe a GIF, TIFF, JPEG, or PNG with compression would be smaller?

File: doc/manual/memory.rst (hunk 14 above)
Comment (Eric89GXL) [inline doc/manual/memory.rst]: And actually I don't think it's true for *most* readers, and even for the ones it is true for, isn't true under all circumstances.

**Comment by Eric89GXL** [comment]
Thanks for tackling this, looks like a useful set of changes

File: doc/tutorials/report.rst (hunk 16 above)
Comment (agramfort) [inline doc/tutorials/report.rst]: have a look at tinypng tool

File: doc/advanced_setup.rst (hunk 8 above)
Comment (jasmainak) [inline doc/advanced_setup.rst]: hmm ... we are not super-consistent with either Python code blocks starting with `>>>` or shell commands starting with `$`. I'd say that the python colors would distinguish them -- so the `>>>` is superfluous

File: doc/tutorials/report.rst (hunk 16 above)
Comment (jasmainak) [inline doc/tutorials/report.rst]: yayy, it's smaller now.

Comment (Eric89GXL) [inline doc/tutorials/report.rst]: 170 kB is still a bit big, do GIF or JPEG do any better?

Comment (jasmainak) [inline doc/tutorials/report.rst]: not by much. Unless I resize. I tried using a weblink and that too doesn't seem to work :\

**Comment by jasmainak** [comment]
I don't think renaming `datasets.rst` breaks any links. The ref at the top of the page was being used for cross-linking anyway

**Comment by agramfort** [comment]
ok good

let us know when it's ready

**Comment by jasmainak** [comment]
ok, I think I'm done. I just learnt how sklearn does the manual linking to images from examples. Would be neat to do. Another time.

File: doc/manual/memory.rst [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    events = mne.find_events(raw)
+    event_id, tmin, tmax = 1, -0.2, 0.5
+    picks = mne.pick_types(raw.info, meg=True, eeg=True, stim=False, eog=True)
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks, baseline=(None, 0), reject=dict(eeg=80e-6, eog=150e-6),
+                        preload=False)
+
+When ``preload=False``, the epochs data is loaded from the disk on-demand. Note that ``preload=False`` for epochs will work even if the ``raw`` object
+has been loaded with ``preload=True``. Preloading is also supported for :func:`mne.read_epochs`.
+
+.. warning:: This comes with a caveat. When ``preload=False``, data rejection based on peak-to-peak thresholds is executed when the data is loaded from disk, *not* when the `raw` object is created.
```
Comment (jaeilepp) [inline doc/manual/memory.rst]: Should this say "'Epochs' object"?

**Comment by jasmainak** [comment]
addressed comment by @jaeilepp . Are we good here?

**Comment by agramfort** [comment]
please squash to 1 commit and let's merge.

squash to remove big file from history

File: doc/advanced_setup.rst [hunk 18]
```diff
@@ -103,6 +103,29 @@ runtime accordingly.
 If you use another Python setup and you encounter some difficulties please
 report them on the MNE mailing list or on github to get assistance.
 
+Installing Mayavi
+^^^^^^^^^^^^^^^^^
+
+Mayavi is only available for Python2.7. If you have Anaconda installed (recommended), the easiest way to install `mayavi` is to do::
+
+    conda install mayavi
```
Comment (Eric89GXL) [inline doc/advanced_setup.rst]: I still think we should add `$` to the `bash` lines even if we haven't been consistent before. I'm okay omitting the `>>>` from Python lines if we have to.

**Comment by jasmainak** [comment]
@Eric89GXL done :) I made the fixes in some other places too.

@agramfort : The screenshot is in one commit already, so no need to squash. You have git ninjas in your lab ;)

**Comment by agramfort** [comment]
thanks @jasmainak

**Pull request merged**